File: javascript/gemini/p124_nr_ver2.js
const args = process.argv.slice(2);

if (args.length !== 1) {
    console.error("Error: Exactly one argument required.");
    process.exit(1);
}

let input = args[0];
console.log(input);

input = input.trimEnd();
console.log(input);

process.exit(0);

File: java/gemini/p124_nr_ver2.java
public class p124_nr_ver2 {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Error: Exactly one argument required.");
            System.exit(1);
        }

        String input = args[0];
        System.out.println(input);

        // Standard trim removes both leading and trailing whitespace
        input = input.trim();
        System.out.println(input);

        System.exit(0);
    }
}